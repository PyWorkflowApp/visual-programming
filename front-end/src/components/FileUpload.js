import React from 'react';
import { Button } from 'react-bootstrap';
import * as API from '../API';

export default class FileUpload extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      input: React.createRef()
    }
  }

   uploadFile = file => {
        const form = new FormData();
        form.append("file", file);
        API.uploadWorkflow(form).then(json => {
            this.props.handleData(json);
        }).catch(err => {
            console.log(err);
        });
        this.setState({input: React.createRef()});
    };

    onFileSelect = e => {
        e.preventDefault();
        if (!this.state.input.current || !this.state.input.current.files) {
          return;
        }

        this.uploadFile(this.state.input.current.files[0]);
    };

    render() {
      return (
          <>
          <input type="file" ref={this.state.input} onChange={this.onFileSelect}
              style={{display: "none"}} />
          <Button size="sm" onClick={() => this.state.input.current.click()}>Load</Button>
          </>
      )
    }
}
