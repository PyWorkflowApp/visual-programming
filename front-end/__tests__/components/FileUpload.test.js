import React from 'react'
import { shallow, mount } from 'enzyme';
import FileUpload from '../../src/components/FileUpload';

global.fetch = jest.fn(() => Promise.resolve({
  json: jest.fn(() => []),
  ok: true
}));

global.console = {log: jest.fn()}

describe('Validates FileUpload', () => {
  it('Displays FileUpload', () => {
    const load = jest.fn(() => []);
    const fileUpload = shallow(<FileUpload
            handleData={load}/>);

    expect(fileUpload).toMatchSnapshot();

    const event = {
      preventDefault: jest.fn(() => [])
    };

    fileUpload.setState({input: { current: { files: ["oneFile"]}}});
    fileUpload.find('input').simulate('change', event);

    expect(event.preventDefault.mock.calls.length).toBe(1);
  });

  it('No file selected', () => {
    const load = jest.fn(() => []);
    const fileUpload = shallow(<FileUpload
            handleData={load}/>);

    const event = {
      preventDefault: jest.fn(() => [])
    };

    fileUpload.setState({input: {}});
    fileUpload.find('input').simulate('change', event);

    expect(event.preventDefault.mock.calls.length).toBe(1);
  });

  it('Load file', () => {
    const load = jest.fn(() => []);
    const fileUpload = shallow(<FileUpload
            handleData={load}/>);

    const click = jest.fn(() => []);

    fileUpload.setState({input: { current: { click: click}}});
    fileUpload.find('Button').simulate('click');

    expect(click.mock.calls.length).toBe(1);
  });

});
