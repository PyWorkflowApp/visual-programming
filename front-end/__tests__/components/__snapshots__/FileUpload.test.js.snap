// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validates FileUpload Displays FileUpload 1`] = `ShallowWrapper {}`;
